system-addon/lib/TopSitesFeed.jsm

```diff
@@ -22,6 +24,13 @@ const DEFAULT_TOP_SITES = [];
 this.TopSitesFeed = class TopSitesFeed {
   constructor() {
     this.lastUpdated = 0;
+    this.dedupe = new Dedupe(this._dedupeKey, this._dedupeCompare);
+  }
+  _dedupeKey(site) {
+    return site.hostname;
+  }
+  _dedupeCompare(storedValue, newValue) {
+    return newValue.isPinned;
```

Comment:
That's what I'm currently working on, realized just as I wanted to request review.